pkg/controller/replication/replication_controller.go

```diff
@@ -478,10 +573,19 @@ func (rm *ReplicationManager) manageReplicas(filteredPods []*api.Pod, rc *api.Re
 		var wg sync.WaitGroup
 		wg.Add(diff)
 		glog.V(2).Infof("Too few %q/%q replicas, need %d, creating %d", rc.Namespace, rc.Name, rc.Spec.Replicas, diff)
+		// needs to set the gvk to generate ownerReference
+		oldKind := rc.Kind
+		oldAPIVersion := rc.APIVersion
+		defer func() {
+			rc.Kind = oldKind
+			rc.APIVersion = oldAPIVersion
+		}()
+		rc.APIVersion = v1.SchemeGroupVersion.String()
+		rc.Kind = "ReplicationController"
```

Comment:
Right, I shouldn't modify the rc. Let me see if there is any elegant way to do the job.